syntax = "proto3"; // PB协议版本
package aiins;

option java_package = "com.example.aiins.proto";

// 获取一系列用户基本信息请求
message UserDataReq {
    repeated int32 uid = 1;
}

message UserData {
    string nickname = 1;
    string username = 2;
    bytes icon = 3;
    int32 uid = 4;
}

// 一系列用户基本信息响应, 根据时间戳返回
message UserDataRsp {
    repeated UserData userData = 1;
}

// 每个用户都有的基本信息
message BasicUserData {
    string nickname = 1;
    string username = 2;
    string password = 3;
    int32 uid = 4;
}

// 本地需要保存的数据, 自己的信息已经隔开
message LocalData {
    repeated Message messages = 3; // 对话信息
}

// 每一个对话信息
message Message {
    int32 from = 1;
    repeated int32 to = 2;
    int32 time = 3;
    string text = 4;
    bytes image = 5;
}